### ISSUE TITLE
Unable to get FCM Token

### ISSUE THREAD

**Issue by eirworks** [OneSignal/OneSignal-Android-SDK#542]
**Description:**

My app unable to receive any notification because Onesignal was unable to get FCM Token.

**Anything else:**
<pre>
E/OneSignal: Unknown error getting FCM Token
    java.lang.IllegalArgumentException: Given String is empty or null
</pre>
It also throws:
<pre>
E/OneSignal: Unknown error getting FCM Token
    java.lang.IllegalStateException: FirebaseApp name ONESIGNAL_SDK_FCM_APP_NAME already exists!
</pre>

Using `OneSignal.getPermissionSubscriptionState()`, I get following: 
`state.subscribed: false`
 `userId: 51273149-*******`
 `pushToken: null`

I have 2 apps with a single backend server. I also put together both app into single Firebase project. So both app has same Project ID, if this makes sense. Also the other app's notification works perfectly.

This is my app's build.gradle:
<pre>
// Play Services
    implementation "com.google.android.gms:play-services-maps:15.0.1"
//    implementation "com.google.android.gms:play-services-gcm:15.0.1"
    implementation "com.google.android.gms:play-services-base:15.0.1"
    implementation "com.google.android.gms:play-services-location:15.0.1"
    implementation "com.google.android.gms:play-services-places:15.0.1"
    implementation 'com.google.android.gms:play-services-auth:15.0.1'
    
    // Firebase
//    implementation "com.google.firebase:firebase-messaging:15.0.2"
    implementation 'com.google.firebase:firebase-auth:15.1.0'
    implementation 'com.google.firebase:firebase-core:15.0.2'
    implementation('com.crashlytics.sdk.android:crashlytics:2.7.1@aar') {
        transitive = true
    }

    // OneSignal
    compile 'com.onesignal:OneSignal:[3.9.0, 3.99.99]'
</pre>

**Comment by eirworks** [comment]
Oh Jon from the OneSignal support asked me to remove `firebase-auth`, which actually fix it, and I wasn't really using it. Thank you for your help. :)

**Comment by jkasten2** [comment]
@eirworks Thanks for the details, good to hear your up and running now!

Just providing a few more details to this for future reference. It seems the 2nd error `FirebaseApp name ONESIGNAL_SDK_FCM_APP_NAME already exists!` just happens as the 1st error `java.lang.IllegalArgumentException: Given String is empty or null` ends up partially initializing a Firebase app.

**Comment by jamesmartins** [comment]
Hi John!
I'm same error! 

Error LogCat:
`
E/OneSignal: Unknown error getting FCM Token
    java.lang.IllegalStateException: Default FirebaseApp is not initialized in this process br.com.rededuque.android. Make sure to call FirebaseApp.initializeApp(Context) first.
        at com.google.firebase.FirebaseApp.getInstance(Unknown Source:53)
        at com.google.firebase.iid.FirebaseInstanceId.getInstance(Unknown Source:0)
        at com.google.firebase.iid.zzao.<init>(Unknown Source:64)
        at com.google.firebase.iid.zzao.<init>(Unknown Source:5)
        at com.google.firebase.iid.FirebaseInstanceId.<init>(Unknown Source:40)
        at com.google.firebase.iid.FirebaseInstanceId.<init>(Unknown Source:9)
        at com.google.firebase.iid.zzaf.create(Unknown Source:10)
        at com.google.firebase.components.zzi.zza(Unknown Source:0)
        at com.google.firebase.components.zzj.get(Unknown Source:4)
        at com.google.firebase.components.zzi.get(Unknown Source:15)
        at com.google.firebase.components.ComponentContainer$$CC.get(Unknown Source:8)
        at com.google.firebase.components.zzg.get(Unknown Source:0)
        at com.google.firebase.components.zzg.zzb(Unknown Source:46)
        at com.google.firebase.FirebaseApp.zzc(Unknown Source:20)
        at com.google.firebase.FirebaseApp.initializeApp(Unknown Source:118)
        at com.onesignal.PushRegistratorFCM.initFirebaseApp(PushRegistratorFCM.java:65)
        at com.onesignal.PushRegistratorFCM.getToken(PushRegistratorFCM.java:50)
        at com.onesignal.PushRegistratorAbstractGoogle.attemptRegistration(PushRegistratorAbstractGoogle.java:97)
        at com.onesignal.PushRegistratorAbstractGoogle.access$100(PushRegistratorAbstractGoogle.java:37)
        at com.onesignal.PushRegistratorAbstractGoogle$1.run(PushRegistratorAbstractGoogle.java:84)
        at java.lang.Thread.run(Thread.java:764)
`

My app/build.gradle:

`
buildscript {
    repositories {
        maven { url 'https://plugins.gradle.org/m2/'}
    }
    dependencies {
        classpath 'gradle.plugin.com.onesignal:onesignal-gradle-plugin:[0.11.0, 0.99.99]'
    }
}
apply plugin: 'com.onesignal.androidsdk.onesignal-gradle-plugin'

repositories {
    maven { url 'https://maven.google.com' }
}
apply plugin: 'com.android.application'

....

dependencies {
    implementation fileTree(dir: 'libs', include: ['*.jar'])

    implementation 'com.android.support:appcompat-v7:27.1.0'
    implementation 'com.android.support.constraint:constraint-layout:1.0.2'
    implementation 'com.android.support:design:27.1.0'
    implementation 'com.android.support:support-v4:27.1.0'

    implementation 'com.google.android.gms:play-services-location:15.0.1'

    implementation 'com.onesignal:OneSignal:[3.9.1, 3.99.99]'

    //Http Web Service Client
    implementation 'com.squareup.okhttp3:okhttp:3.2.0'

    testImplementation 'junit:junit:4.12'
    androidTestImplementation 'com.android.support.test:runner:1.0.1'
    androidTestImplementation 'com.android.support.test.espresso:espresso-core:3.0.1'

}

`

**Comment by jkasten2** [comment]
@eirworks We just released a fix for this yesterday in 3.9.3.
https://github.com/OneSignal/OneSignal-Android-SDK/releases/tag/3.9.3

Can you try re-syncing your gradle to pull in the latest version? Thanks.

**Comment by soshial** [comment]
got this error with these versions:

```
compile "com.onesignal:OneSignal:3.10.2"
plugins {
    id 'com.onesignal.androidsdk.onesignal-gradle-plugin' version '0.12.1'
}
classpath 'com.google.gms:google-services:4.1.0'
```

**Comment by jkasten2** [comment]
@soshial The `ONESIGNAL_SDK_FCM_APP_NAME already exists!` error was fixed in `3.9.3`. Can you share the exact error you are getting in a new issue? Thanks.

**Comment by cloudtenlabs** [comment]
> Hi John!
> I'm this same error!
> 
> Error LogCat:
> 
> `E/OneSignal: Unknown error getting FCM Token java.lang.IllegalStateException: Default FirebaseApp is not initialized in this process br.com.rededuque.android. Make sure to call FirebaseApp.initializeApp(Context) first. at com.google.firebase.FirebaseApp.getInstance(Unknown Source:53) at com.google.firebase.iid.FirebaseInstanceId.getInstance(Unknown Source:0) at com.google.firebase.iid.zzao.<init>(Unknown Source:64) at com.google.firebase.iid.zzao.<init>(Unknown Source:5) at com.google.firebase.iid.FirebaseInstanceId.<init>(Unknown Source:40) at com.google.firebase.iid.FirebaseInstanceId.<init>(Unknown Source:9) at com.google.firebase.iid.zzaf.create(Unknown Source:10) at com.google.firebase.components.zzi.zza(Unknown Source:0) at com.google.firebase.components.zzj.get(Unknown Source:4) at com.google.firebase.components.zzi.get(Unknown Source:15) at com.google.firebase.components.ComponentContainer$$CC.get(Unknown Source:8) at com.google.firebase.components.zzg.get(Unknown Source:0) at com.google.firebase.components.zzg.zzb(Unknown Source:46) at com.google.firebase.FirebaseApp.zzc(Unknown Source:20) at com.google.firebase.FirebaseApp.initializeApp(Unknown Source:118) at com.onesignal.PushRegistratorFCM.initFirebaseApp(PushRegistratorFCM.java:65) at com.onesignal.PushRegistratorFCM.getToken(PushRegistratorFCM.java:50) at com.onesignal.PushRegistratorAbstractGoogle.attemptRegistration(PushRegistratorAbstractGoogle.java:97) at com.onesignal.PushRegistratorAbstractGoogle.access$100(PushRegistratorAbstractGoogle.java:37) at com.onesignal.PushRegistratorAbstractGoogle$1.run(PushRegistratorAbstractGoogle.java:84) at java.lang.Thread.run(Thread.java:764)`
> 
> My app/build.gradle:
> 
> `
> buildscript {
> repositories {
> maven { url '[https://plugins.gradle.org/m2/'}](https://plugins.gradle.org/m2/'%7D)
> }
> dependencies {
> classpath 'gradle.plugin.com.onesignal:onesignal-gradle-plugin:[0.11.0, 0.99.99]'
> }
> }
> apply plugin: 'com.onesignal.androidsdk.onesignal-gradle-plugin'
> 
> repositories {
> maven { url 'https://maven.google.com' }
> }
> apply plugin: 'com.android.application'
> 
> android {
> // My settings
> }
> dependencies {
> implementation fileTree(dir: 'libs', include: ['*.jar'])
> 
> ```
> implementation 'com.android.support:appcompat-v7:27.1.0'
> implementation 'com.android.support.constraint:constraint-layout:1.0.2'
> implementation 'com.android.support:design:27.1.0'
> implementation 'com.android.support:support-v4:27.1.0'
> 
> implementation 'com.google.android.gms:play-services-location:15.0.1'
> 
> implementation 'com.onesignal:OneSignal:[3.9.1, 3.99.99]'
> 
> //Http Web Service Client
> implementation 'com.squareup.okhttp3:okhttp:3.2.0'
> 
> testImplementation 'junit:junit:4.12'
> androidTestImplementation 'com.android.support.test:runner:1.0.1'
> androidTestImplementation 'com.android.support.test.espresso:espresso-core:3.0.1'
> ```
> 
> }
> `
